core/injector/application_injector.hpp

```diff
@@ -280,13 +284,39 @@ namespace kagome::injector {
 
       auto &context = injector.template create<boost::asio::io_context &>();
       auto extrinsic_tcp_version = boost::asio::ip::tcp::v4();
-      api::ListenerImpl::Configuration listener_config{
+      api::HttpListenerImpl::Configuration listener_config{
           boost::asio::ip::tcp::endpoint{extrinsic_tcp_version, rpc_port}};
       auto &&http_session_config =
           injector.template create<api::HttpSession::Configuration>();
 
-      initialized = std::make_shared<api::ListenerImpl>(
-          context, listener_config, http_session_config);
+      initialized = std::make_shared<api::HttpListenerImpl>(
+		  context, listener_config, http_session_config);
+      return initialized.value();
+    };
+
+    // jrpc api listener (over Websockets) getter
+    auto get_jrpc_api_ws_listener = [](const auto &injector,
+                                    uint16_t rpc_port) -> sptr<api::WsListenerImpl> {
+      static auto initialized =
+          boost::optional<sptr<api::WsListenerImpl>>(boost::none);
+      // listener is used currently only for extrinsic api
+      // if other apis are required, need to
+      // implement lambda creating corresponding api service
+      // where listener is initialized manually
+      // in this case listener should be bound
```

Comment:
I think this comment is not valid anymore as listener is used for more than just extrinsic api

Reply:
We can safely remove it from this and previous lambda